### ISSUE TITLE
Export selected objects

### ISSUE THREAD

**Issue by HansBosma** [archimatetool/archi#955]
Is it possible to export a selection of objects? It would be nice to have an export of objects in some folder (so export all objects within folder). Or does an jArchi script already exists for this functionality?

**Comment by Phillipus** [comment]
Hi, what export format did you have in mind?

**Comment by HansBosma** [comment]
Oh yes sorry, just the ordinary csv-export. And I am already happy with just the 'elements' csv format.

**Comment by Phillipus** [comment]
Hi, the CSV export implementation exports all elements, relations, and properties for the whole model so it would be difficult to re-write that code for a specific use case. There may be a jArchi script out there that does this, though.

**Comment by HansBosma** [comment]
Yes I know, I was hoping I could have / find that script 'out there'... Or maybe that it is simple to change the standard script to work only on a selection of objects but I am not able to change the script in this way.

**Comment by HansBosma** [comment]
The examples jArchi script (is the same as the standard export functionality in Archi I presume?)

**Comment by Phillipus** [comment]
> The examples jArchi script

I guess you mean the "Export to Excel" one. If so, you could change this part:

```js
$("element").each(function(e) {
    excel.set(0, 0, current_row, e.name);
    excel.set(0, 1, current_row, e.documentation);
    excel.set(0, 2, current_row, e.id);
    excel.set(0, 3, current_row, e.type);
    excel.set(0, 4, current_row, e.specialization == null ? "" : e.specialization);
    
    current_row++;
});
```

Instead of using `$("element")` as the selector ("all elements") you could change it to get the currently selected folder and all elements in that folder.

If you get stuck, it's probably best to discuss on the [Archi Forum](https://forum.archimatetool.com/index.php?board=5.0), as there are some jArchi experts who might offer some advice.

**Comment by Phillipus** [comment]
Assuming that you select a folder in the models tree, the jArchi `selection` will give you the selected folder. Then you can use `children` to get the child elements:

```
selection.children().each(function(e) {
```

(Reference - https://github.com/archimatetool/archi-scripting-plugin/wiki/jArchi-Collection)

**Comment by HansBosma** [comment]
Magnificent! I did not know where to look, and it works now. Thank's a lot!!

**Comment by Phillipus** [comment]
You're welcome. If you do get into jArchi more, please do discuss on the [Archi Forum](https://forum.archimatetool.com/index.php?board=5.0)! :-)
